### ISSUE TITLE
Use `python -m pip install` in all examples

### ISSUE THREAD

**Issue by dbeatty10** [fishtown-analytics/docs.getdbt.com#3468]
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Any page that includes `pip install`. (Dozens and dozens of pages!)

Use something like the following to find them:
```shell
git grep "pip install" website/docs/
```

### What part(s) of the page would you like to see updated?

Just like the [examples](https://pip.pypa.io/en/stable/cli/pip_install/#examples) in the pip documentation, replace all instances of `pip install` with `python -m pip install`

### Additional information

The rationale is explained well by others:
- https://snarky.ca/why-you-should-use-python-m-pip/
- https://adamj.eu/tech/2020/02/25/use-python-m-pip-everywhere/

#### TLDR
- Python and pip might point to different versions of Python, and the recommended syntax allow you to be very purposeful and explicit in what Python interpreter you want to be used/affected. Otherwise a [5 hour fight](https://github.com/pypa/pip/issues/3164#issuecomment-1104524768) with issues is possible.

This is considered a universal best practice because it works regardless of the situation whereas without `python -m` it may or may not work as intended. Always using the best practice helps teach other people about it.

**Comment by runleonarun** [comment]
Thanks for opening this @dbeatty10! I marked it high priority as we are not supporting a best practice in the docs.

**Comment by runleonarun** [comment]
Addressed by https://github.com/dbt-labs/docs.getdbt.com/pull/4498
